### PR TITLE
Add checkbox to enable/disable SFX

### LINKED ISSUE #275: [2.8.6] Add a separate sfx checkmark
me and CrazyJC literally forgot to do that but it's really important. That was supposed to be there at launch but now I'm reminded by some nerd that WebAO actually has it and 2.8 doesn't.

I'm crying. I'm literally shakening right now. This is a disaster, 2.8 is cancelled, everyone uninstall cuz we don't have an sfx checkmark we need to fix this NOW PLEASE
изображение

### PR DESCRIPTION
**Pull request by skyedeving**
Fixes #275

This add this entry to your theme to place the checkbox below the casing one

; SFX Enable/disable
sfx = 200, 580, 80, 21

### CODE REVIEW

File: src/courtroom.cpp [hunk 1]
```diff
@@ -383,6 +387,7 @@ Courtroom::Courtroom(AOApplication *p_ao_app) : QMainWindow()
   connect(ui_additive, SIGNAL(clicked()), this, SLOT(on_additive_clicked()));
   connect(ui_guard, SIGNAL(clicked()), this, SLOT(on_guard_clicked()));
   connect(ui_casing, SIGNAL(clicked()), this, SLOT(on_casing_clicked()));
+  connect(ui_sfx, &QCheckBox::clicked, [this]() {ui_ic_chat_message->setFocus();});
```
Comment (oldmud0) [inline src/courtroom.cpp:390]: Not the right way of doing this if you want to use new-style connections.
```suggestion
  connect(ui_sfx, &QCheckBox::clicked, ui_ic_chat_message, &QWidget::setFocus);
```

File: src/courtroom.cpp [hunk 2]
```diff
@@ -4090,6 +4098,7 @@ bool Courtroom::effects_dropdown_find_and_set(QString effect)
 
 QString Courtroom::get_char_sfx()
 {
+  if (!ui_sfx->isChecked()) return "0";
```
Comment (oldmud0) [inline src/courtroom.cpp:4101]:
```suggestion
  if (!ui_sfx->isChecked())
    return "1";
```
Consistent with default return value of [`get_sfx_name`](https://github.com/AttorneyOnline/AO2-Client/blob/849f91d991bf0e95d579df268e9ab358e09b1ac1/src/text_file_functions.cpp#L771).

**Review by oldmud0 — CHANGES_REQUESTED** [review]

File: src/courtroom.cpp (hunk 1 above)
Comment (skyedeving) [reply src/courtroom.cpp]: Neat

**Comment by Crystalwarrior** [comment]
Yo thanks for handling this, however I already have a different non standard solution ready to commit that instead adds a "Nothing" option to the sfx drop down instead. The option behaves very similar to "Default" tho "Nothing" blocks all sfx without blocking anims

I couldn't commit due to 403 error but I should be back sometime this week to commit it

**Pull request closed without merging**